db-engine/src/main/java/com/splicemachine/db/impl/sql/compile/BinaryListOperatorNode.java

```diff
@@ -472,4 +473,24 @@ public int getOuterJoinLevel() {
     public void setOuterJoinLevel(int level) {
         outerJoinLevel = level;
     }
+
+    public ValueNode replaceIndexExpression(ResultColumnList childRCL) throws StandardException {
+        ValueNodeList newList = (ValueNodeList) getNodeFactory().getNode(
+                C_NodeTypes.VALUE_NODE_LIST,
+                getContextManager());
+        for (Object leftItem : leftOperandList) {
+            newList.addValueNode(((ValueNode) leftItem).replaceIndexExpression(childRCL));
+        }
+        leftOperandList = newList;
+        return this;
```

Comment:
We need to check the rightOperandList also.

Reply:
Done.